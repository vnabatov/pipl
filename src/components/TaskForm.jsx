import React from 'react'
import styled from 'styled-components'

const Form = styled.div`
width: 40%;
padding: 1rem;
`
const Button = styled.input`
margin-right: 0.5rem;
`

const fields = ['id', 'summary', 'teamName', 'related', 'sp', 'story']

export default ({ form, updateTask, deleteTask, clearForm }) => {
  const updateForm = (field, value) => {
    form[field] = value
  }
  return <Form className='hero'>
    {fields.map(field => <div className='field is-horizontal'>
      <div className='field-label is-small'>
        <label className='label'>{field}</label>
      </div>
      <div className='field-body'>
        <div className='field'>
          <p className='control'>
            <input className='input is-small' type='text' value={form[field]} onChange={e => updateForm(field, e.target.value)} />
          </p>
        </div>
      </div>
    </div>)}

    <div className='field is-horizontal'>
      <div className='field-label is-normal' />
      <div className='field-body'>
        <div className='field'>
          <p className='control'>
            <Button className='button is-primary' type='button' value='Save' onClick={updateTask} />
            <Button className='button is-warning' type='button' value='Delete' onClick={() => deleteTask()} />
            <Button className='button' type='button' value='Clear' onClick={clearForm} />
          </p>
        </div>
      </div>
    </div>
  </Form>
}
