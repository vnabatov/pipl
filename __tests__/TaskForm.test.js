import React from 'react'
import { create } from 'react-test-renderer'
import TaskForm from '../src/components/TaskForm'
import AppContext from '../src/AppContext'

describe('TaskForm', () => {
  test('Matches the snapshot', () => {
    const taskForm = create(<AppContext.Provider value={{
      deleteTask: () => {},
      updateTask: () => {},
      clearForm: () => {},
      dependendTasks: {},
      taskPostionsCache: {},
      isCompact: false
    }}
    ><TaskForm />
    </AppContext.Provider>)
    expect(taskForm.toJSON()).toMatchSnapshot()
  })
})
