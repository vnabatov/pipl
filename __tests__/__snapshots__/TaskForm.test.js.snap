exports[`TaskForm Matches the snapshot 1`] = `
<div
  className="sc-bdVaJa jAdfzN">
  <div
    className="field is-horizontal">
    <div
      className="field-label is-small">
      <label
        className="label">
        id
      </label>
    </div>
    <div
      className="field-body">
      <div
        className="field">
        <div
          className="control">
          <input
            className="input is-small"
            defaultValue={undefined}
            onChange={[Function]}
            type="text" />
        </div>
      </div>
    </div>
  </div>
  <div
    className="field is-horizontal">
    <div
      className="field-label is-small">
      <label
        className="label">
        description
      </label>
    </div>
    <div
      className="field-body">
      <div
        className="field">
        <div
          className="control">
          <input
            className="input is-small"
            onChange={[Function]}
            type="text"
            value={undefined} />
        </div>
      </div>
    </div>
  </div>
  <div
    className="field is-horizontal">
    <div
      className="field-label is-small">
      <label
        className="label">
        sp
      </label>
    </div>
    <div
      className="field-body">
      <div
        className="field">
        <div
          className="control">
          <input
            className="input is-small"
            onChange={[Function]}
            type="text"
            value={undefined} />
        </div>
      </div>
    </div>
  </div>
  <div
    className="field is-horizontal">
    <div
      className="field-label is-small">
      <label
        className="label">
        summary
      </label>
    </div>
    <div
      className="field-body">
      <div
        className="field">
        <div
          className="control">
          <input
            className="input is-small"
            onChange={[Function]}
            type="text"
            value={undefined} />
        </div>
      </div>
    </div>
  </div>
  0
  0
  0
  <div
    className="field is-horizontal">
    <div
      className="field-label is-normal" />
    <div
      className="field-body">
      <div
        className="field">
        <p
          className="control">
          <input
            className="button is-primary sc-bwzfXH ksNfkq"
            onClick={[Function]}
            type="button"
            value="Save" />
          <input
            className="button is-warning sc-bwzfXH ksNfkq"
            onClick={[Function]}
            type="button"
            value="Delete" />
          <input
            className="button sc-bwzfXH ksNfkq"
            onClick={[Function]}
            type="button"
            value="Close" />
        </p>
      </div>
    </div>
  </div>
</div>
`;
